### PR TITLE
[TEST ONLY][NO MERGE] Surface Leak Testing Builds

### PR DESCRIPTION
**Pull request by kd-11**
Duplicate of https://github.com/RPCS3/rpcs3/pull/9107 but has options merger which is a future PR. That change fixes the crashing originally reported in the other PR.

### CODE REVIEW

File: rpcs3/Emu/RSX/GL/GLTextureCache.h [hunk 1]
```diff
@@ -350,6 +350,38 @@ namespace gl
...
+						[&src_rtt](const auto& region)
+						{
+							return (region.source == src_rtt);
+						}
+					))
+					{
+						continue;
+					}
+
+					rtt->inherit_surface_contents(src_rtt);
```
Comment (Megamouse) [inline rpcs3/Emu/RSX/GL/GLTextureCache.h:380]:
```suggestion
					if (std::none_of(rtt->old_contents.begin(), rtt->old_contents.end(),
						[&src_rtt](const auto& region)
						{
							return (region.source == src_rtt);
						}
					))
					{
						rtt->inherit_surface_contents(src_rtt);
					}
```

File: rpcs3/Emu/RSX/VK/VKTextureCache.h [hunk 2]
```diff
@@ -319,6 +313,38 @@ namespace vk
...
+						[&src_rtt](const auto& region)
+						{
+							return (region.source == src_rtt);
+						}
+					))
+					{
+						continue;
+					}
+
+					rtt->inherit_surface_contents(src_rtt);
```
Comment (Megamouse) [inline rpcs3/Emu/RSX/VK/VKTextureCache.h:343]:
```suggestion
					if (std::none_of(rtt->old_contents.begin(), rtt->old_contents.end(),
						[&src_rtt](const auto& region)
						{
							return (region.source == src_rtt);
						}
					))
					{
						rtt->inherit_surface_contents(src_rtt);
					}
```

File: rpcs3/Emu/RSX/Common/surface_utils.h [hunk 3]
```diff
@@ -504,6 +504,44 @@ namespace rsx
 			}
 		}
 
+		template <typename T>
+		std::pair<bool, bool> inherit_surface_contents(T* other)
```
Comment (Megamouse) [inline rpcs3/Emu/RSX/Common/surface_utils.h:508]: maybe add a comment about the return pair values

**Comment by kd-11** [comment]
No longer required, the root cause of the broken visuals was discovered.

**Pull request closed without merging**